com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs

```diff
@@ -555,29 +585,67 @@ public override void Setup(ScriptableRenderContext context, ref RenderingData re
 
-#if ENABLE_VR && ENABLE_XR_MODULE
-                if (cameraData.xr.enabled)
+                if (RenderingUtils.RTHandleNeedsReAlloc(m_DepthTexture, depthDescriptor, false))
                 {
-                    activeColorRenderTargetId = new RenderTargetIdentifier(activeColorRenderTargetId, 0, CubemapFace.Unknown, -1);
-                    activeDepthRenderTargetId = new RenderTargetIdentifier(activeDepthRenderTargetId, 0, CubemapFace.Unknown, -1);
+                    m_DepthTexture?.Release();
+                    m_DepthTexture = RTHandles.Alloc(depthDescriptor, filterMode: FilterMode.Point, wrapMode: TextureWrapMode.Clamp, name: "_CameraDepthTexture");
                 }
-#endif
 
-                ConfigureCameraTarget(activeColorRenderTargetId, activeDepthRenderTargetId);
+                CommandBuffer cmd = CommandBufferPool.Get();
+                cmd.SetGlobalTexture(m_DepthTexture.name, m_DepthTexture.nameID);
+                context.ExecuteCommandBuffer(cmd);
+                CommandBufferPool.Release(cmd);
             }
 
-            bool hasPassesAfterPostProcessing = activeRenderPassQueue.Find(x => x.renderPassEvent == RenderPassEvent.AfterRenderingPostProcessing) != null;
-
-            if (mainLightShadows)
-                EnqueuePass(m_MainLightShadowCasterPass);
+            // Allocate normal texture if used
+            if (requiresDepthPrepass && renderPassInputs.requiresNormalsTexture)
+            {
+                var normalDescriptor = cameraTargetDescriptor;
+                normalDescriptor.depthBufferBits = 0;
+                // Never have MSAA on this depth texture. When doing MSAA depth priming this is the texture that is resolved to and used for post-processing.
+                normalDescriptor.msaaSamples = 1;// Depth-Only pass don't use MSAA
+                // Find compatible render-target format for storing normals.
+                // Shader code outputs normals in signed format to be compatible with deferred gbuffer layout.
+                // Deferred gbuffer format is signed so that normals can be blended for terrain geometry.
+                if (this.actualRenderingMode == RenderingMode.Deferred)
+                    normalDescriptor.graphicsFormat = m_DeferredLights.GetGBufferFormat(m_DeferredLights.GBufferNormalSmoothnessIndex); // the one used by the gbuffer.
+                else if (RenderingUtils.SupportsGraphicsFormat(GraphicsFormat.R8G8B8A8_SNorm, FormatUsage.Render))
+                    normalDescriptor.graphicsFormat = GraphicsFormat.R8G8B8A8_SNorm; // Preferred format
+                else if (RenderingUtils.SupportsGraphicsFormat(GraphicsFormat.R16G16B16A16_SFloat, FormatUsage.Render))
+                    normalDescriptor.graphicsFormat = GraphicsFormat.R16G16B16A16_SFloat; // fallback
+                else
+                    normalDescriptor.graphicsFormat = GraphicsFormat.R32G32B32A32_SFloat; // fallback
```

Comment:
can this setup code be moved in the DepthNormalOnlyPass setup for better readability? (I see a similar pattern for other passes too, in general if we can move any pass specific code into the actual pass would be better to keep UniversalRenderer cleaner)

Reply:
Do you mean getting the specific format from `DepthNormalOnlyPass` or allocating the `RTHandle`?

In this PR I had to move ownership of the Textures out of the passes and into the renderers.

Reply:
I mean any logic specific to the specific pass, so the descriptor only. (or a pass specific utility method to initialize the descriptor for the pass), leaving ownership of the textures to the renderers